src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngProperty.cs

```diff
@@ -27,7 +27,14 @@ public CngProperty(string name, byte[]? value, CngPropertyOptions options)
             _value = value.CloneByteArray();
         }
 
-        internal CngProperty(string name, ReadOnlySpan<byte> value, CngPropertyOptions options)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CngProperty"/> struct.
+        /// </summary>
+        /// <param name="name">The property name to initialize.</param>
+        /// <param name="value">The property value to initialize.</param>
+        /// <param name="options">A bitwise combination of the enumeration values that specify how the property is stored.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is <see langref="null"/>.</exception>
+        public CngProperty(string name, ReadOnlySpan<byte> value, CngPropertyOptions options)
```

Comment:
What I care about is that there's a test that says a CngProperty built on `(byte[])null` is the same as one built on `(ReadOnlySpan<byte>)(byte[])null`, and similarly for `Array.Empty<byte[]>()`.  If that equivalence is met, I don't particularly care what this line of code looks like :smile:.

If a CngProperty based on `null` is basically useless (CNG will never return one, and CNG will never accept one... it can only be used within application code for some random purpose) then my position probably changes to "the null span and the empty span should be the same".  But I want to start with the tests.  Particularly if null and empty are both valid inputs to CNG but mean different things (e.g. if null means "erase this property")

Reply:
I still don't see tests around this behavior.

There should be a test showing that setting a property to `(byte[])null` (via the byte[] ctor) and `default(ReadOnlySpan<byte>)` (via the new ctor) do the same thing.

There should also be a test (same one? different one? surprise me) showing that `Array.Empty<byte>()` (old ctor) and `Array.Empty<byte>().AsSpan()` (new ctor) do the same thing.  They might even do the same thing as `null`.

What sort of things are the same?

* Does HasProperty give the same answer for the same value in both the old and new ctor when assigning a property that the key doesn't already have?
* Does HasProperty give the same answer for the same value in both the old and new ctor when overwriting an existing (non-empty) value?
* Does GetProperty give identical answers for the same value in both the old and new ctor?
* Anything else that feels relevant, though their might not actually be "anything else" since there's not a lot of API surface here.